rules/naming/src/Rector/Assign/RenameVariableToMatchGetMethodNameRector.php

```diff
@@ -0,0 +1,118 @@
+use Rector\Core\RectorDefinition\CodeSample;
+use Rector\Core\RectorDefinition\RectorDefinition;
+use Rector\Naming\Naming\ExpectedNameResolver;
+use Rector\NodeTypeResolver\Node\AttributeKey;
+
+/**
+ * @see \Rector\Naming\Tests\Rector\Assign\RenameVariableToMatchGetMethodNameRector\RenameVariableToMatchGetMethodNameRectorTest
+ */
+final class RenameVariableToMatchGetMethodNameRector extends AbstractRector
+{
+    /**
+     * @var ExpectedNameResolver
+     */
+    private $expectedNameResolver;
+
+    public function __construct(ExpectedNameResolver $expectedNameResolver)
+    {
+        $this->expectedNameResolver = $expectedNameResolver;
+    }
+
+    public function getDefinition(): RectorDefinition
+    {
+        return new RectorDefinition('Rename variable to match get method name', [
+            new CodeSample(
+                <<<'PHP'
+class SomeClass {
+    public function run()
+    {
+        $a = $this->getRunner();
+    }
+}
+PHP
+,
+                <<<'PHP'
+class SomeClass {
+    public function run()
+    {
+        $runner = $this->getRunner();
+    }
+}
+PHP
+            ),
+        ]);
+    }
+
+    /**
+     * @return string[]
+     */
+    public function getNodeTypes(): array
+    {
+        return [Assign::class];
+    }
+
+    /**
+     * @param Assign $node
+     */
+    public function refactor(Node $node): ?Node
+    {
+        if ($node->expr instanceof ArrowFunction) {
+            return null;
+        }
+
+        /** @var MethodCall|StaticCall|FuncCall $callNode */
+        $callNode = $this->betterNodeFinder->findFirst($node->expr, function ($node) {
+            return $node instanceof MethodCall || $node instanceof StaticCall || $node instanceof FuncCall;
+        });
+
+        if ($callNode === null) {
+            return null;
+        }
+
+        $newName = $this->expectedNameResolver->resolveForMethodCall($callNode);
+        if ($newName === null || $this->isName($node, $newName)) {
+            return null;
+        }
+
+        /** @var Node\FunctionLike $classLikeNode */
+        $functionLikeNode = $this->betterNodeFinder->findFirstParentInstanceOf($node, Node\FunctionLike::class);
+        if ($functionLikeNode !== null) {
+            $nodes = $functionLikeNode->getStmts();
+        } else {
+            $functionLikeNode = $node->getAttribute(AttributeKey::PARENT_NODE);
+            $nodes = [];
+            /** @var Node\Stmt\Expression $expression */
+            while ($expression = $functionLikeNode->getAttribute(AttributeKey::NEXT_NODE)) {
+                $nodes[] = $expression;
+            }
+        }
+
+        /** @var Variable $variableNode */
+        $variableNode = $node->var;
+        $originalName = $variableNode->name;
+
+        $this->traverseNodesWithCallable($nodes, function ($node) use ($originalName, $newName): void {
+            if ($this->isVariableName($node, $originalName)) {
+                $node->name = $newName;
+            }
+        });
```

Comment:
This is probably variable rename? Could be standalone method too, just to show in the name